File: vestro-sales/src/types/product.ts
export type ProductCategory = 'accessories' | 'audio' | 'desk' | 'mobile' | 'network' | 'work';

export type Product = {
    id: string;
    name: string;
    description: string;
    category: ProductCategory;
    price: number;
    stock: number;
    image_url: string;
    is_active: boolean;
    created_at: string;
    updated_at: string;
    discount_percent?: number | null;
    discounted_price?: number | null;
};

export type DiscountScope = 'all' | 'category' | 'product';

export type Discount = {
    id: string;
    scope: DiscountScope;
    category?: string | null;
    productId?: string | null;
    productName?: string | null;
    percentage: number;
    startDate: string;
    endDate?: string | null;
    createdAt: string;
};

export type DiscountCreateInput = {
    scope: DiscountScope;
    category?: string | null;
    productId?: string | null;
    percentage: number;
    startDate: string;
    endDate?: string | null;
};

export type ProductCreateInput = Omit<Product, 'id' | 'created_at' | 'updated_at' | 'is_active'> & {
    is_active?: boolean;
};

File: api-vestro/app/schemas/product.py
from typing import Literal

from pydantic import BaseModel

ProductCategory = Literal["accessories", "audio", "desk", "mobile", "network", "work"]


class Product(BaseModel):
    id: str
    name: str
    description: str | None = None
    category: str
    price: float
    stock: int
    image_url: str | None = None
    is_active: bool
    created_at: str
    updated_at: str


class ProductCreateRequest(BaseModel):
    name: str
    description: str | None = None
    category: ProductCategory
    price: float
    stock: int = 0
    image_url: str | None = None


class ProductUpdateRequest(BaseModel):
    name: str | None = None
    description: str | None = None
    category: ProductCategory | None = None
    price: float | None = None
    stock: int | None = None
    image_url: str | None = None
    is_active: bool | None = None
